api/bin/chainflip-cli/src/main.rs

```diff
@@ -412,3 +423,36 @@ fn generate_keys(json: bool, path: Option<PathBuf>, seed_phrase: Option<String>)
 
 	Ok(())
 }
+
+#[test]
+fn test_flip_to_redemption_amount() {
+	assert_eq!(flip_to_redemption_amount(None), RedemptionAmount::Max);
+	assert_eq!(flip_to_redemption_amount(Some(0.0)), RedemptionAmount::Exact(0));
+	assert_eq!(flip_to_redemption_amount(Some(-1000.0)), RedemptionAmount::Exact(0));
+	assert_eq!(
+		flip_to_redemption_amount(Some(199995.0)),
+		RedemptionAmount::Exact(199995000000000000000000)
+	);
+
+	assert_eq!(
+		flip_to_redemption_amount(Some(123456789.000001)),
+		RedemptionAmount::Exact(123456789000001000000000000)
+	);
+
+	assert_eq!(
+		flip_to_redemption_amount(Some(69420.123456)),
+		RedemptionAmount::Exact(69420123456000000000000)
+	);
+
+	// specifying more than the allowed precision ignores the extra precision
+	assert_eq!(
+		flip_to_redemption_amount(Some(69420.123456789)),
+		RedemptionAmount::Exact(69420123456000000000000)
```

Comment:
This is a bit weird - I know it makes little difference but I'd prefer if we either rounded to the nearest decimal, or were just super strict and refused anything with more than the supported precision. 

ie. 69420.123456789 should either Error or give 69420.123457.